extension/src/openvic-extension/singletons/MapItemSingleton.cpp

```diff
@@ -0,0 +1,229 @@
+	OV_BIND_METHOD(MapItemSingleton::get_capital_positions);
+	OV_BIND_METHOD(MapItemSingleton::get_crime_icons);
+	OV_BIND_METHOD(MapItemSingleton::get_rgo_icons);
+}
+
+MapItemSingleton* MapItemSingleton::get_singleton() {
+	return singleton;
+}
+
+MapItemSingleton::MapItemSingleton() {
+	ERR_FAIL_COND(singleton != nullptr);
+	singleton = this;
+}
+
+MapItemSingleton::~MapItemSingleton() {
+	ERR_FAIL_COND(singleton != this);
+	singleton = nullptr;
+}
+
+// Get the billboard object from the loaded objects
+
+GFX::Billboard const* MapItemSingleton::get_billboard(std::string_view name, bool error_on_fail) const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, nullptr);
+	DefinitionManager const* definition_manager = &game_singleton->get_definition_manager();
+	ERR_FAIL_NULL_V(definition_manager, nullptr);
+
+	GFX::Billboard const* billboard =
+		definition_manager->get_ui_manager().get_cast_object_by_identifier<GFX::Billboard>(name);
+
+	if (error_on_fail) {
+		//ERR_FAIL_NULL_V_MSG(billboard, nullptr, vformat("Failed to find billboard \"%s\"", Utilities::std_to_godot_string(name)));
+		ERR_FAIL_NULL_V_MSG(billboard, nullptr, "Failed to find billboard inner");
+	}
+
+	return billboard;
+}
+
+// repackage the billboard object into a godot dictionnary for the Billboard manager to work with
+bool MapItemSingleton::add_billboard_dict(std::string_view name, TypedArray<Dictionary>& billboard_dict_array) {
+
+	static const StringName name_key = "name";
+	static const StringName texture_key = "texture";
+	static const StringName scale_key = "scale";
+	static const StringName noOfFrames_key = "noFrames";
+	//TODO: might need font, font_size, and offset keys in the future
+
+
+	GFX::Billboard const* billboard = get_billboard(name,false);
+
+	ERR_FAIL_NULL_V_MSG(
+		billboard, false, "Failed to find billboard"
+	);
+
+	Dictionary dict;
+
+	dict[name_key] = Utilities::std_to_godot_string(billboard->get_name());
+	dict[texture_key] = Utilities::std_to_godot_string(billboard->get_texture_file());
+	dict[scale_key] = billboard->get_scale().to_float();
+	dict[noOfFrames_key] = billboard->get_no_of_frames();
+
+	billboard_dict_array.push_back(dict);
+
+	return true;
+}
+
+
+//get an array of all the billboard dictionnaries
+TypedArray<Dictionary> MapItemSingleton::get_billboards(){
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, {});
+
+	DefinitionManager const* definition_manager = &game_singleton->get_definition_manager();
+	ERR_FAIL_NULL_V(definition_manager, TypedArray<Dictionary>());
+
+	TypedArray<Dictionary> ret;
+
+	for (std::unique_ptr<GFX::Object> const& obj : definition_manager->get_ui_manager().get_objects()) {
+		if (obj->is_type<GFX::Billboard>()) {
+			add_billboard_dict(obj->get_name(), ret);
+		}
+	}
+	
+	return ret;
+}
+
+//dont use this, it includes water provinces
+int MapItemSingleton::get_province_count(){
```

Comment:
Better to always be explicit with the integer type size, so `int32_t` here and for `get_capital_count`. These should also be changed in the hpp file